README.md

```diff
@@ -3,6 +3,112 @@ python-krux-boto
 
 Krux Python class built on top of [Krux Stdlib](https://staticfiles.krxd.net/foss/docs/pypi/krux-stdlib/) for interacting with [Boto](http://boto.readthedocs.org/en/latest/) and [Boto3](http://boto3.readthedocs.org/en/latest/index.html)
 
+CRITICAL NOTE about versions!
+-----------------------------
+
+Version 0.0.7 went through a major refactoring work and thus is not compatible with applications written with `krux-boto` version 0.0.6. The mechanism to fetch AWS credential is changed and may cause your application to use a different credentials than intended. Therefore, version 0.0.7 **should not be used**. Please use either 0.0.6 or 1.0.0 version.
+
+### Updating from 0.0.6 to 1.0.0
+
+In version 0.0.6, `krux_boto.Boto` object took an `argparse.ArgumentParser` object as an optional parameter for the constructor. This approach has been abandoned. `krux_boto.Boto` object now expects 4 parameters listed below. Therefore, following change is required to get your application working with version 1.0.0.
+
+```python
+
+# Version 0.0.6
+
+import krux.cli
+from krux_boto import Boto, add_boto_cli_arguments
+
+class Application(krux.cli.Application):
+
+    def __init__(self, name='krux-my-boto-script'):
+        super(Application, self).__init__(name=name)
+        
+        self.boto = Boto(
+            parser=self.parser,
+            logger=self.logger,
+            stats=self.stats,
+        )
+        
+    def add_cli_arguments(self, parser):
+        super(Application, self).add_cli_arguments(parser)
+        
+        add_boto_cli_arguments(parser)
+
+```
+```python
+
+# Version 1.0.0
+
+import krux.cli
+from krux_boto import Boto, add_boto_cli_arguments
+
+class Application(krux.cli.Application):
+
+    def __init__(self, name='krux-my-boto-script'):
+        super(Application, self).__init__(name=name)
+        
+        # Notice the change in constructor parameters
+        self.boto = Boto(
+            log_level=self.args.boto_log_level,
+            access_key=self.args.boto_access_key,
+            secret_key=self.args.boto_secret_key,
+            region=self.args.boto_region,
+            logger=self.logger,
+            stats=self.stats,
+        )
+        
+    def add_cli_arguments(self, parser):
+        super(Application, self).add_cli_arguments(parser)
+        
+        add_boto_cli_arguments(parser)
+
+```
+
+If you were not setting any of the optional parameters, use `krux_boto.Boto.get_boto()` function.
```

Comment:
Oh, you mention it here - i'm a little confused when i should use one or the other, and how i could mix/match.

Can this be distilled to a single code example, or a hard & fast rule?